pkg/controller/build/reconciler.go

```diff
@@ -975,6 +979,14 @@ func (b *buildReconciler) syncMachineOSConfig(ctx context.Context, mosc *mcfgv1.
 			return fmt.Errorf("could not create new or reuse existing MachineOSBuild for MachineOSConfig %q: %w", mosc.Name, err)
 		}
 
+		pool, err := b.mcfgclient.MachineconfigurationV1().MachineConfigPools().Get(ctx, mosc.Spec.MachineConfigPool.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		// Set the owner ref of the mosc to the pool
+		oref := metav1.NewControllerRef(pool, mcfgv1.SchemeGroupVersion.WithKind("MachineConfigPool"))
```

Comment:
question (non-blocking): Do we want to have the MachineOSConfig be deleted if the pool is deleted? I ask because the precedent for MachineConfigs is that only rendered MachineConfigs would get deleted when the pool is; component MachineConfigs, such as ones provided by a cluster admin, are not automatically deleted.

Reply:
If the pool is deleted but the machineoscofnig remains, won't we run into a bunch of error logs about MOSC not being able to be applied because pool doesn't exist. I have not tested that case, so not sure what exactly happens but that is what I was thinking when setting the pool as the owner of the mosc.

Reply:
We would run into those, yes. However, I think that might be preferable because the MOSC is user-created. And so would any MachineConfigPools other than the control-plane and worker pools.

Reply:
Makes sense, I have removed the owner from the MOSC so nothing will automatically clear it out anymore.